platform/admin/admin-configuration-configinstaller/src/test/java/org/codice/ddf/admin/configuration/ConfigurationInstallerTest.java

```diff
@@ -0,0 +1,175 @@
+import static org.osgi.service.cm.ConfigurationEvent.CM_UPDATED;
+
+import java.io.File;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.cm.ConfigurationEvent;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConfigurationInstallerTest {
+    private static final String FELIX_FILENAME_PROP = "felix.fileinstall.filename";
+
+    // Will be tracked in the installer's map after init()
+    private static final String PID_001 = "001";
+
+    // Will NOT be tracked in the installer's map after init()
+    private static final String PID_002 = "002";
+
+    @Mock
+    private Configuration configuration;
+
+    @Mock
+    private ConfigurationAdmin configurationAdmin;
+
+    @Mock
+    private ConfigurationEvent configurationEvent;
+
+    private ConfigurationInstaller installer;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private File file1a;
+
+    private File file1b;
+
+    @Before
+    public void before() throws Exception {
+        temporaryFolder.create();
+        when(configurationEvent.getType()).thenReturn(CM_UPDATED);
+
+        file1a = temporaryFolder.newFile();
+        file1b = temporaryFolder.newFile();
+
+        // Ensure init takes the last value for a particular key
+        Configuration mockConfig1a = createMockConfig(PID_001, file1a.toURI());
+        Configuration mockConfig1b = createMockConfig(PID_001, file1b.toURI());
+        Configuration mockConfig2 = createMockConfig(PID_002, null);
+
+        Configuration[] configs = new Configuration[] {mockConfig1a, mockConfig1b, mockConfig2};
+        when(configurationAdmin.listConfigurations(null)).thenReturn(configs);
+
+        installer = new ConfigurationInstaller(configurationAdmin);
+        installer.init();
+    }
+
+    @Test
+    public void testInit() throws Exception {
+        Map<String, File> pidFileMap = installer.getPidFileMap();
+        assertThat(pidFileMap.entrySet(), hasSize(1));
+        assertThat(pidFileMap.get(PID_001), is(file1b));
+    }
+
+    @Test
+    public void testUpdatedEventNotTrackingWontTrack() throws Exception {
+        configuration = createMockConfig(PID_002, null);
+
+        when(configurationEvent.getPid()).thenReturn(PID_002);
+        when(configurationAdmin.getConfiguration(PID_002, null)).thenReturn(configuration);
+
+        installer.configurationEvent(configurationEvent);
+
+        Map<String, File> pidFileMap = installer.getPidFileMap();
+        assertThat(pidFileMap.entrySet(), hasSize(1));
+        assertThat(pidFileMap.get(PID_001), is(file1b));
+        assertThat(file1b.exists(), is(true));
```

Comment:
Because if we are not tracking and won't start tracking that config, we should ensure the initial state of the filesystem was preserved. In this case, that file1b was not removed.